dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -168,9 +172,22 @@ function Catalog(props: ICatalogProps) {
   // Only one search filter can be set
   const searchFilter = propsFilter[filterNames.SEARCH]?.toString() || "";
   const reposFilter = filters[filterNames.REPO]?.join(",") || "";
+
+  const clusterUpdate = useRef(cluster);
+  const namespaceUpdate = useRef(namespace);
+  const reposFilterUpdate = useRef(reposFilter);
+  const searchFilterUpdate = useRef(searchFilter);
+
+  // detect changes in cluster/ns/repos/search but do NOT update them
+  // until a resetRequestCharts has been called to avoid colissions
+  // fetchCharts is only called in observeBorder
   useEffect(() => {
-    fetchCharts(cluster, namespace, reposFilter, page, size, searchFilter);
-  }, [fetchCharts, cluster, namespace, reposFilter, page, size, searchFilter]);
+    clusterUpdate.current = cluster;
+    namespaceUpdate.current = namespace;
+    reposFilterUpdate.current = reposFilter;
+    searchFilterUpdate.current = searchFilter;
+    resetRequestCharts();
+  }, [resetRequestCharts, cluster, namespace, reposFilter, searchFilter]);
```

Comment:
If I understand the problem here, if you are in page 3 and then you click in a repo filter, you want to reset the page counter, right? (So you request the first page again). Or you are trying some performance improvement?

What I would do, to try to simplify this a bit (and potentially avoid the `refs`), would be to have the `page` in the component state (rather than in redux), and everytime there is a change in the filters, reset the page. I am not sure if that'd trigger two requests though (one for the page: 0 and old filters and one for page: 0 and the new filters or just the second).

Reply:
I used the `useRef` beacuse of I explained in https://github.com/kubeapps/kubeapps/pull/2264#discussion_r561084979.

Mmm, but, although we store `page` in the component state, we will still have it as one of the dependencies of the `useEffect`, so, regardless of how it is stored, it would cause a refresh whenever it is updated, wouldn't it?

I don't know, anyway I will give it a try for sure.

Reply:
yes, but that's what we want, no? If we are at `page: 2, repo: ""` and we change it to `repo: "bitnami` and **at the same time** (this is what I don't know if it's possible) `page: 0` we would do a single `fetchChart` with `page: 0, repo: "bitnami"`. Isn't that correct?